pkg/releases/releases.go

```diff
@@ -0,0 +1,79 @@
+package releases
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/google/go-github/v43/github"
+)
+
+const (
+	githubNamespace  = "authzed"
+	githubRepository = "spicedb"
+
+	pageSize = 25
+)
+
+// Release represents a release of SpiceDB.
+type Release struct {
+	// Version is the version of the release.
+	Version string
+
+	// PublishedAt is when the release was published, in UTC.
+	PublishedAt time.Time
+
+	// ViewURL is the URL at which the release can be viewed.
+	ViewURL string
+}
+
+// GetLatestRelease returns the latest release of SpiceDB, as reported by the GitHub API.
+func GetLatestRelease(ctx context.Context) (*Release, error) {
+	client := github.NewClient(nil)
+	var current *Release
+
+	// Load the releases from GitHub in a paginated fashion (to not overwhelm the API).
```

Comment:
Did you consider using the API specific to get the latest release instead of iterating over all releases?

https://docs.github.com/en/rest/releases/releases#get-the-latest-release

Reply:
My concern there was if we published a "fix" release; say we're at 1.6.0 and we publish 1.5.7 as a fix afterwards for some reason. Will GitHub now report 1.5.7 as the "latest" release?